### PR TITLE
Pull request for to-do-app

### PR DESCRIPTION
**Pull request by wintan1418**
This web app puts into practice the given instructions as to how it should be built.
Users can create several activities or tasks to do and save them for reminders using this web app.
it works perfectly fine.

### CODE REVIEW

**Review by trekab — COMMENTED** [review]
#### Invalid! :stop_sign: 
Hi @wintan1418! Unfortunately, I have to invalidate your code review request because you didn't set up linting as respected from [this guide](https://github.com/microverseinc/linters-config/tree/master/javascript).

__Please make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__[Treasure Kabareebe](https://trekab.github.io)__

**Comment by wintan1418** [comment]
I am so sorry, I just added it now, I asked for another review,my bad

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  -  Follows gitwork-flow. 💯
  -  Design looks good. 💯

- ### To Improve
  
  - [x] [MANDATORY] Make sure that the project can be created without creating a to-do(task). which means the project can also be created(mandatory) or deleted(optional) separately.  and then when creating the task, you add it to an existing project. 
  - [x] [MANDATORY] Default project should also have default tasks.
  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Comment by wintan1418** [comment]
thank you

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+const projectNameList = (list) => {
+  projects.forEach((project) => list.push(project.name));
+  return list;
+};
+
+const validateForm = (title, date, priority) => !(title === '' || date === '' || priority === 'Choose...');
+
+const setAlert = (alert, status) => {
+  alert.style.display = 'block';
```
Comment (rindrajosia) [inline src/index.js]: [OPTIONAL] Keep all the styling logic in CSS files . Here for example, you can create class called `.block `in your CSS file then add this class name attribute with JavaScript

File: src/factory.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default function projectFactory(name) {
```
Comment (rindrajosia) [inline src/factory.js]: [OPTIONAL] Use `arrow functions` instead of `the function keyword`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good job by using localStorage :+1: 
- Webpack configuration is perfect :100: 

To improve :factory_worker: [REQUIRED]:

- [x] Please work on the previous TSE requirements before asking for new code review.

Nice to have [OPTIONAL]: 

- [x] Please use this file as linter configuration : [.github/workflows/linters.yml](https://github.com/microverseinc/linters-config/blob/master/javascript/.github/workflows/linters.yml)
- [x] As you are using ES6 syntax, use `arrow functions` instead of `the function keyword`
- [x] Keep all the styling logic in CSS files  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Happy coding
Rindra Josia

**Comment by wintan1418** [comment]
thanks for the reviews, I have done the requested change.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @wintan1418 👋

**You've done pretty well here fixing a few issues raised in the previous review. 👌**

However, let's make a couple of changes.

- [ ] You are still yet to fulfill these requirements: There should be a default project this means when the page first loads the user should see a default project and when creating a todo-item he can choose to add that todo item to the default project. The user should also be able to add more projects and can assign to do tasks to any of these new projects or to the default project. Hoping this is explanatory enough.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  displayProjects(projects);
+};
+
+// Parses the form input
+const forminput = () => {
+  // const title = document.querySelector('#inputtitle').value.trim();
+  // const date = document.querySelector('#inputdate').value;
+  // const description = document.querySelector('#inputdescription').value.trim();
+  // const note = document.querySelector('#inputnote').value;
+  // const priority = document.querySelector('#inputpriority').value;
```
Comment (rukundoeric) [inline src/index.js]: remove this commented-out code

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  -  Follows gitwork-flow. 💯
  -  Design looks okay. 💯 

- ### To Improve
  
  - [ ] [MANDATORY] You still haven't implemented the main mandatory requirements from all of the previous reviews. kindly check the previous reviewer's comment and implement the required changes. in case you still don't understand what you have to do, use the following `live-demo` links as a reference.
    - Refs:
       - https://raw.githack.com/rukundoeric/todo-list/todo-list/dist/index.html
       - https://kumaramitt.github.io/TODO-List/
       - https://raw.githack.com/janis-jenny/ToDo-List/todoApp/dist/index.html   
  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by Forison — CHANGES_REQUESTED** [review]
# Status::Changes Required❌

# 👋 @wintan1418 

I can see you have come a long way,but there is still more room for improvement😇.

* [ ] Currently the `default project` has been added, however  **I am unable to create a todo** nor **other project with different names either than default**. 
Note: The project should be such that there is a default project that stores todo that are saved under the project name default. There should be also a way for users to create a project name either than default and store todo under this new project name. Consequently we should be able to view `todos` that are associated to each project name created in the project.
Please let us take our time to implement these features, as mentioned earlier feel free to contact any TSE for assistance, you can find us by our name on slack.

#### Kindly make the changes and submit a re-review😎

Happy coding👨🏾‍💻

Addo Forison

**Comment by wintan1418** [comment]
please these reviews are quite confusing, I have done all that you asked, and it works fine at my end, and tse keeps complaining, I keep damaging my code, and I have checked with other people, they got approved without this persistent issue, please kindly allow me to move on to the next project, it is really demeaning

**Comment by rukundoeric** [comment]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Implemented some of the feedback from the previous revious. 💯 

- ### To Improve
  
  - [ ] [MANDATORY] You still don't get it, the project should be independent. which means, I should be able to create a project separately. but in your case, the project can only be created when creating a task that is not correct. kindly follow these simple steps to fix this:
    -  Above the form you already have, create a new form with one `text-field`:(you can call it project_name) and just save/create `button`. this form will be used to create projects.
    - Then finally, convert the following field into a selector:
      Screen Shot 2021-05-20 at 20 07 21
      you can use the same field you used for Priority. this `select` will list all projects including default so that when creating a task, the user will select which project he/she wants to assign the task to.

  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work.  💻 👀 ✨
- I like the design.
- The code structure is alright, but still needs some work.
## However, there are some changes required 👀 🧰 🔨 🔧
- The interface is very confusing. The first button is for "create new project", but then it gets toggled to "modify task" when I click on the "edit task" icons from the tasks. I think it would be better to use the second button for task edition, since it's the one corresponding to the task data form.
- When I try to create a new project, a task gets created instead. Please bear in mind that the first button should be to create projects, and the second button to create tasks inside the projects.
- Please find a way to select a project before creating a task, so that the tasks get created inside a project selected by the user.

### Submit these changes and request a new code review! 😄 👍

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier)🧑‍💻🗨️🧑‍💻
# Happy Coding! 🧠 ⌨️ 🖱️ 🖥️

**Review by umairarshadbutt — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
**It is the 8th review but there are still issues with the edit pf task. I will go ahead and mark this as complete.**
- You can merge to master before fix the issues and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
HAPPY CODING :computer:
**[Umair Arshad (Sodtware Engineer at MTBC](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Pull request merged**